README.md
# Styled Maps add-on for Concrete5
An easy to use Google maps add-on that allows the user to customise their map to suit their website theme.

## Features
* Built in theme options (basic)
* Custom theme option via JSON (advanced)
* Add custom map marker images and set marker size
* Map width & height settings
* Ability to enable/disable Scroll wheel, Satellite view and Street view options
* Configurable zoom settings
* Info window pop-up with rich text editor and options to display on map load or when map marker is clicked

## Install
1. Download and unzip this package into your site's packages/ directory.
2. Login to your site as an administrator.
3. Find the "Add Functionality" page in your dashboard.
4. Find this package in the list of packages awaiting installation.
5. Click the "install" button.


CHANGELOG.txt
 Date (d/m/Y) | Version | Description / Comment
 ================================================
 
 3/5/2016 , 0.9.0 , PRB Submission
 ------------------------------------------------
 6/5/2016 , 0.9.1 , Added selectable themes, Map marker sizing control,
                    Custom code validation (check code is between []),
                    Split form into tabbed areas, added styling to form hints.
------------------------------------------------
10/5/2016 , 0.9.2 , Added popup Info Window functionality, added block header info / help,
                    added form tool tips, fixed missing t() function in view.php,
                    replaced unescape()funtion with decodeURI() in auto.js,
                    updated links to Google maps API to remove an issue where
                    undeclared libraries in the URL would result in duplicate .js files being included
                    when the block was added to the page.     
------------------------------------------------
11/5/2016 , 0.9.3 , Added option to select default info window state (onclick / onload ),
                    Added js to hide custom code options unless 'Custom (Advanced)' has been selected from Standard Themes dropdown menu. 
------------------------------------------------
24/5/2016 , 1.0.0 , PRB Approved
------------------------------------------------
10/6/2016 , 1.1.0 , Added Satellite control feature, Added Street view control feature, Updated scroll wheel option from checkbox to select.
------------------------------------------------
27/6/2016 , 1.2.0 , Added Google maps API key input to Styled Maps form. Required by Google as of 22 June 2016.
------------------------------------------------
29/7/2016 , 1.2.1 , Fixed bug caused by WYSIWYG editor adding unwanted line breaks to Info window resulting in maps not rendering.
------------------------------------------------
30/8/2016 , 1.2.2 , Remapped database schema to support C5 versions from 5.7.1 up to 5.7.4.2 (Skipped in this version).
------------------------------------------------
30/8/2016 , 1.2.3 , Update versioning to match current version for all C5 installs 5.7.5 and up.

blocks/styled_maps/auto.js
(function () {
    "use strict";
 
    window.C5GMaps = {
        
        pacTimer: null,
        
        init: function () {
            if (!window.C5GMaps.isMapsPresent()) {
                $('head').append($(decodeURI("%3Cscript src='https://maps.googleapis.com/maps/api/js?key=ADD_KEY_HERE&libraries=places&callback=window.C5GMaps.setupAutocomplete' type='text/javascript'%3E%3C/script%3E")));
            } else {
                window.C5GMaps.setupAutocomplete();
            }
        },
        
        isMapsPresent: function () {
            if (typeof google === 'object'
                    && typeof google.maps === 'object'
                    && typeof google.maps.places === 'object') {
                return true;
            }
            return false;
        },
        
        setupAutocomplete: function () {
            
            var input = $("#ccm-google-map-block-location > input[id=location]").get(0),
                autocomplete = new google.maps.places.Autocomplete(input),
                note = $("#ccm-google-map-block-location > input[id=location]").get(0);
            input.onchange = function () {
                this.className = 'notfound form-control ccm-input-text';
            };
            
            // Keeps the autocomplete list visible above modal dialogue
            window.C5GMaps.pacTimer = setInterval(function () {
                var cntr = $('.pac-container'), locBox = $('#location');
                cntr.css('z-index', '2000');
                if (locBox.length === 0) {
                    clearInterval(window.C5GMaps.pacTimer);
                    cntr.remove();
                }
            }, 250);

            google.maps.event.addDomListener(input, 'keydown', function(e) {
                if (e.keyCode == 13) {
                    e.preventDefault();
                }
            });
            google.maps.event.addListener(autocomplete, 'place_changed', function () {

                var place = autocomplete.getPlace();
                if (!place.geometry) {
                    // Inform the user that the place was not found and return.
                    input.className = 'notfound';
                    note.innerHTML = 'The place you entered could not be found.';
                    return;
                } else {
                    $('#ccm-google-map-block-location > input[id=latitude]').val(place.geometry.location.lat());
                    $('#ccm-google-map-block-location > input[id=longitude]').val(place.geometry.location.lng());
                    input.className = 'form-control ccm-input-text';
                    note.innerHTML = '';
                }
                
            });
            
            
        }
    };

}());


blocks/styled_maps/view.php
<?php     defined('C5_EXECUTE') or die("Access Denied.");

$c = Page::getCurrentPage();
if ($c->isEditMode()) { ?>
	<div class="ccm-edit-mode-disabled-item" style="width: <?php     echo $width; ?>; height: <?php     echo $height; ?>">
		<div style="padding: 80px 0px 0px 0px"><?php     echo t('Styled Maps disabled in edit mode.')?></div>
	</div>
<?php      } else { ?>
	<?php      if( strlen($title)>0){ ?><h3><?php     echo $title?></h3><?php      } ?>
	<div id="googleMapCanvas<?php     echo $unique_identifier?>" class="googleMapCanvas" style="width: <?php     echo $width; ?>; height: <?php     echo $height; ?>"></div>
<?php      } ?>



<?php    
/*
    Note - this goes in here because it's the only way to preserve block caching for this block. We can't
    set these values through the controller
*/
?>

<script type="text/javascript">
    function googleMapInit<?php     echo $unique_identifier?>() {
        try{
            var latlng = new google.maps.LatLng(<?php     echo $latitude?>, <?php     echo $longitude?>);
            var mapOptions = {
                zoom: <?php     echo $zoom?>,
                center: latlng,                     
                <?php   
                if($satelliteControl == 'yes') { ?>           
                mapTypeControl: true,
                <?php   
                }
                else {
                ?>
                mapTypeControl: false,
                <?php   
                }
                ?>
                mapTypeId: google.maps.MapTypeId.ROADMAP,
                <?php    if($streetControl == 'yes') { ?>
                streetViewControl: true,                
                <?php   
                }
                else {
                ?>
                streetViewControl: false,
                <?php   
                }
                if($scrollwheel == 'yes') { ?>
                scrollwheel: true,
                <?php   
                }
                else {
                ?>
                scrollwheel: false,
                <?php   
                }
                ?>
                zoomControl: true,
                <?php    
                    // Check for custom styling code (Advanced options)
		    if($snazzycode != "" && $themehex == "custom")
		    {
		        echo 'styles: '.$snazzycode;
		    }
		    // else if 'Default' theme selected from dropdown, Default to standard Google map styling
		    elseif($themehex == 'google') {
		    	echo '';
		    }
		    // else display the chosen theme (Basic option)
		    else
		    { ?>
		    
		        
		        styles: [
                                  {
                                    "featureType": "all",
                                    "elementType": "all",
                                    "stylers": [
                                                  {
                                                    "hue": "#<?php     echo $themehex; ?>"
                                                  },
                                                  {
                                                    <?php    
                                                        // If basic theme is Greyscale then set the saturation to -100
                                                        if($themehex == '222222'){
                                                    ?>    
                                                            "saturation": "-100"
                                                    <?php    
                                                    	}
                                                    	else { 
                                                    ?>
                                                    	    "saturation": "0"	
                                                    <?php    
                                                        }
                                                    ?>    	            
                                                  }
                                               ]
                                  }
                                ]
               <?php     } ?>
            };
            var map = new google.maps.Map(document.getElementById('googleMapCanvas<?php     echo $unique_identifier?>'), mapOptions);
            var customicon = {
                               <?php     if($fID > 0) { $f = \File::getByID($fID); ?>
                               url: <?php     echo "'".$f->getURL()."'"; ?>,
                               scaledSize : new google.maps.Size(<?php     echo $markerwidth.",".$markerheight; ?>)
                               <?php     } ?> 
                             };
            var marker = new google.maps.Marker({
                position: latlng,
                <?php    
                if($fID > 0)
                {
                    echo "map: map, icon: customicon,";
                }
                else
                {
                    echo 'map: map,';
                }
                ?>                                                              
            });
            
            <?php    
            // Add info window popup if text available
            if($infotext != "") {
            ?>
                 var contentString = "<?php     echo $infotext; ?>";

                 var infowindow = new google.maps.InfoWindow({
                     content: contentString
                 });
                 <?php     
                 // Check info window display state (Show window or wait for user to click marker)
                 if($infostate == 1) {
                 ?>
                     marker.addListener('click', function() {
                     infowindow.open(map, marker);
                     });
                 <?php    
                 }
                 else { ?>
                     infowindow.open(map, marker);
                     
                     marker.addListener('click', function() {
                     infowindow.open(map, marker);
                     });
                 <?php    
                 }
             } // End if $infotext
             ?>
                                   
        }catch(e){
            $("#googleMapCanvas<?php     echo $unique_identifier?>").replaceWith(e.message)}
    }
    $(function() {
        var t;
        var startWhenVisible = function (){
            if ($("#googleMapCanvas<?php     echo $unique_identifier?>").is(":visible")){
                window.clearInterval(t);
                googleMapInit<?php     echo $unique_identifier?>();
                return true;
            }
            return false;
        };
        if (!startWhenVisible()){
            t = window.setInterval(function(){startWhenVisible();},100);
        }
    });
</script>

blocks/styled_maps/controller.php
<?php    

namespace Concrete\Package\StyledMaps\Block\StyledMaps;

use Page;
use Concrete\Core\Block\BlockController;
use Core;

class Controller extends BlockController
{
    protected $btTable = 'btStyledMaps';
    protected $btInterfaceWidth = "520";
    protected $btInterfaceHeight = "780";
    protected $btCacheBlockRecord = true;
    protected $btCacheBlockOutput = true;
    protected $btCacheBlockOutputOnPost = true;
    protected $btCacheBlockOutputForRegisteredUsers = false;
    protected $btDefaultSet = 'multimedia';

    public $apikey = "";
    public $title = "";
    public $location = "";
    public $latitude = "";
    public $longitude = "";
    public $scrollwheel = "";
    public $satelliteControl = "";
    public $streetControl = "";
    public $zoom = 14;
    public $themehex = "";
    public $snazzycode = "";
    public $infotext = "";
    public $infostate = true;
    public $fID = "";

    /**
     * Used for localization. If we want to localize the name/description we have to include this.
     */
    public function getBlockTypeDescription()
    {
        return t("Add a styled Google map to your website.");
    }

    public function getBlockTypeName()
    {
        return t("Styled Maps");
    }
    
    public function getBlockTypeHelp()
    {
        return t("<p>Styled maps allows you to place a Google map anywhere on your site and apply a custom theme and location marker.</p>
                  <p>From beginners to experienced users, Styled maps makes customising your Google maps both quick and easy.</p>
                  <p>Use one of the standard themes or take total control and add an advanced / custom map styling by utilising the great resouces available at Snazzy Maps.</p>
                  <p>Add your own custom map marker for limitless design options.</p>
                  <p>For further information and documentation see <a href='http://www.cfive.advite.com.au/documentation/styled-maps-docs' target='_blank'>here</a>.</p>");
    }

    public function validate($args)
    {
        $error = Core::make('helper/validation/error');

        if (empty($args['location']) || $args['latitude'] === '' || $args['longtitude'] === '') {
            $error->add(t('You must select a valid location.'));
        }

        if (!is_numeric($args['zoom'])) {
            $error->add(t('Please enter a zoom number from 0 to 21.'));
        }
        
        // Update v0.9.1
        // Make sure map marker width & height is numeric only
        if (!is_numeric($args['markerwidth']) || !is_numeric($args['markerheight'])) {
             $error->add(t('Please make sure the map marker width and heights are a numerical value.'));
        }
        
        // Update v0.9.1
        // Add validation for Snazzy maps custom code to make sure any pasted code is contained between square brackets []
        if(!empty($args['snazzycode'])) {
           if(substr($args['snazzycode'], 0,1) != '[' || substr($args['snazzycode'], -1) != ']') {
            $error->add(t('Please make sure your custom code is correct and is contained between square brackets [].'));
           } 
        }
        
        // Update v0.9.1
        // Make sure Custom code has been added to the form if "Custom" theme option was selected from the dropdown menu
        if(empty($args['snazzycode']) && $args['themehex'] == 'custom') {
            $error->add(t('Please add custom theme code or choose a standard theme from the list.'));
        }
        
        // Check API Key has been supplied
        if(empty($args['apikey'])) {
            $error->add(t('Please enter your Google maps API key.'));
        }

        // Check for errors 
        if ($error->has()) {
            return $error;
        }
                
    } 


    public function view()
    {
        $this->set('unique_identifier', Core::make('helper/validation/identifier')->getString(18));
        $this->set('bID', $this->bID);
        $this->set('apikey', $this->apikey);
        $this->set('title', $this->title);
        $this->set('location', $this->location);
        $this->set('latitude', $this->latitude);
        $this->set('longitude', $this->longitude);
        $this->set('zoom', $this->zoom);
        $this->set('scrollwheel', $this->scrollwheel);
        $this->set('satelliteControl', $this->satelliteControl);
        $this->set('streetControl', $this->streetControl);
        $this->set('themehex', $this->themehex);
	$this->set('snazzycode', $this->snazzycode);
	$this->set('infotext', $this->infotext);
	$this->set('infostate', $this->infostate);
	$this->set('fID', $this->fID);
    }
    
    public function registerViewAssets($outputContent = '')
    {
        // If no API key (initial setup) use default key
        if(!$this->apikey) {
            $this->requireAsset('javascript', 'jquery');
            $this->addFooterItem(
               '<script src="https://maps.googleapis.com/maps/api/js?key=ADD_KEY_HERE&libraries=places"></script>'
            );
        }
        // otherwise access Google maps via user defined API key
        else {
            $api = $this->apikey;
            $this->requireAsset('javascript', 'jquery');
            $this->addFooterItem(
               '<script src="https://maps.googleapis.com/maps/api/js?key='.$api.'&libraries=places"></script>'
            );
        }    
    }

    public function save($data)
    {
        $data += array(
           'apikey' => '',
           'title' => '',
           'location' => '',
           'zoom' => -1,
           'latitude' => 0,
           'longitude' => 0,
           'width' => null,
           'scrollwheel' => '',
           'satelliteControl' => '',
           'streetControl' => '',
           'themehex' => '',
	   'snazzycode' => '',
	   'infotext' => '',
	   'infostate' => 0,
	   'fID' => 0,
        );
        $args['apikey'] = trim($data['apikey']);
        $args['title'] = trim($data['title']);
        $args['location'] = trim($data['location']);
        $args['zoom'] = (intval($data['zoom']) >= 0 && intval($data['zoom']) <= 21) ? intval($data['zoom']) : 14;
        $args['latitude'] = is_numeric($data['latitude']) ? $data['latitude'] : 0;
        $args['longitude'] = is_numeric($data['longitude']) ? $data['longitude'] : 0;
        $args['width'] = $data['width'];
        $args['height'] = $data['height'];
        $args['scrollwheel'] = trim($data['scrollwheel']);
        $args['satelliteControl'] = trim($data['satelliteControl']);
        $args['streetControl'] = trim($data['streetControl']);
        $args['markerwidth'] = $data['markerwidth'];
        $args['markerheight'] = $data['markerheight'];
        $args['themehex'] = trim($data['themehex']);
	$args['snazzycode'] = trim($data['snazzycode']);
	$args['infotext'] = str_replace('"', "'", $data['infotext']);
	$args['infotext'] = trim($args['infotext']);
	$args['infotext'] = preg_replace( "/\r|\n/", "", $args['infotext']);
	$args['infostate'] = $data['infostate'] ? 1 : 0;
	$args['fID'] = is_numeric($data['fID']) ? $data['fID'] : 0;
        parent::save($args);
    }
}


blocks/styled_maps/form_setup_html.php
<?php     defined('C5_EXECUTE') or die("Access Denied.");
$al = Core::make('helper/concrete/asset_library');
print Core::make('helper/concrete/ui')->tabs(array(
      array('map-settings', t('Map Settings'), true),
      array('map-styling', t('Map Styling')),
      array('map-info', t('Add Info Window'))
));
?>
<div id="ccm-tab-content-map-settings" class="ccm-tab-content">
<div class="ccm-google-map-block-container row">       

    <div class="col-xs-12"> 
        <div class="form-group">
            <label class="control-label" for="apikey" name="apikey">Google Maps API key
                <i class="launch-tooltip fa fa-question-circle" title="<?php     echo t('Paste your Google Maps API Key here. (Required as of 22 June 2016)'); ?>"></i>
            </label>
            <?php     echo $form->text('apikey', $mapObj->apikey);?>
        </div>
        
        <div class="form-group">
            <label class="control-label" for="title" name="title">Map Title (Optional)
                <i class="launch-tooltip fa fa-question-circle" title="<?php     echo t('If you would like to have a title (heading) above your map then add it here.'); ?>"></i>
            </label>
            <?php     echo $form->text('title', $mapObj->title);?>
        </div>

        <div id="ccm-google-map-block-location" class="form-group">
            <label class="control-label" for="location" name="location">Location
                <i class="launch-tooltip fa fa-question-circle" title="<?php     echo t('Enter a location. This will become the focus of the map / map marker.') ?>"></i>
            </label>
            <?php     echo $form->text('location', $mapObj->location);?>
            <?php     echo $form->hidden('latitude', $mapObj->latitude);?>
            <?php     echo $form->hidden('longitude', $mapObj->longitude);?>
        </div>
            
        <div class="col-xs-12">
              <div id="block_note" class="alert alert-info" role="alert">
                 <?php     echo t('Start typing the name or address of a location then choose the correct option from the list.')?>
              </div>             
            <div id="map-canvas"></div>	
        </div>
    </div>

    <div class="col-xs-4">    
        <div class="form-group">
            <label class="control-label" for="zoom" name="zoom">Zoom
                <i class="launch-tooltip fa fa-question-circle" title="<?php     echo t('Set the zoom level of your map. Select a number between 0 & 21 with 0 being world view and 21 being street level.') ?>"></i>
            </label>
            <?php    
                $zoomArray = array();
                for($i=0;$i<=21;$i++) {
                    $zoomArray[$i] = $i;
                }
            ?>
            <?php     echo $form->select('zoom', $zoomArray, $mapObj->zoom);?>
        </div>
    </div>

    <div class="col-xs-4">	
        <div class="form-group"> 
            <label class="control-label" for="width" name="width">Map Width
                <i class="launch-tooltip fa fa-question-circle" title="<?php     echo t('Set the width of your map here.') ?>"></i>
            </label>
            <div class="input-group">
                <span class="input-group-addon"><i class="fa fa-arrows-h"></i></span>
                <?php     if(is_null($width) || $width == 0) {$width = '100%';};?>
                <?php     echo $form->text('width', $width);?>
            </div>
        </div>
    </div>

    <div class="col-xs-4">
        <div class="form-group">
             <label class="control-label" for="height" name="height">Map Height
                <i class="launch-tooltip fa fa-question-circle" title="<?php     echo t('Set the height of your map here.') ?>"></i>
            </label>
            <div class="input-group">
                <span class="input-group-addon"><i class="fa fa-arrows-v"></i></span>
                <?php     if(is_null($height) || $height == 0) {$height = '400px';};?>
                <?php     echo $form->text('height', $height); ?>
            </div>
        </div>
    </div> 

    <div class="col-xs-12 col-md-4">
        <div class="form-group">
            <label class="control-label" for="scrollwheel" name="scrollwheel">                
                 Scroll Wheel
                 <i class="launch-tooltip fa fa-question-circle" title="<?php     echo t('Select \'Yes\' if you would like your map to zoom in and out by moving the scroll wheel.') ?>"></i>
            </label>
            <?php    
               $scrollArray = array('yes'=>'Yes','no'=>'No');              
               echo $form->select('scrollwheel', $scrollArray, $mapObj->scrollwheel);
            ?>
        </div>
    </div> 
    
    <div class="col-xs-12 col-md-4">
        <div class="form-group">
            <label class="control-label" for="satelliteControl" name="satelliteControl">                
                 Satellite View
                 <i class="launch-tooltip fa fa-question-circle" title="<?php     echo t('Add Satellite view controls to your map. Users can toggle a satellite view of your map.') ?>"></i>
            </label>
            <?php    
               $satArray = array('yes'=>'Yes','no'=>'No');              
               echo $form->select('satelliteControl', $satArray, $mapObj->satelliteControl);
            ?>
        </div>
    </div> 
    
    <div class="col-xs-12 col-md-4">
        <div class="form-group">
            <label class="control-label" for="streetControl" name="streetControl">                
                 Street View
                 <i class="launch-tooltip fa fa-question-circle" title="<?php     echo t('Add street view controls to your map. Users can toggle a street view of your map') ?>"></i>
            </label>
            <?php    
                $streetArray = array('yes'=>'Yes','no'=>'No');              
                echo $form->select('streetControl', $streetArray, $mapObj->streetControl);
             ?>
        </div>
    </div>  
       
  </div><!-- / Row -->
</div> <!-- / map-settings tab -->    

<div id="ccm-tab-content-map-styling" class="ccm-tab-content">
<div class="ccm-google-map-block-container row">
    <div class="col-xs-12">
	    <div class="form-group">
	        <?php     $f = \File::getByID($mapObj->fID);?>		
		<label class="control-label" for="fID" name="fID">Add a custom map marker (optional)
                    <i class="launch-tooltip fa fa-question-circle" title="<?php     echo t('Upload or select an image from the file manager that will be used to replace the default Google map marker.') ?>"></i>
                </label>
		<?php     echo $al->file('fID', 'fID', t('Choose image file'), $f);?>
	    </div>
	</div>
	
	<div class="col-xs-6">	
        <div class="form-group">            
            <label class="control-label" for="markerwidth" name="markerwidth">Custom Map Marker Width
                <i class="launch-tooltip fa fa-question-circle" title="<?php     echo t('If you added a custom map marker image above, then you can set a custom Width in pixels for it here. * Enter a number only') ?>"></i>
            </label>
            <div class="input-group">
                <span class="input-group-addon"><i class="fa fa-arrows-h"></i></span>
                <?php     if(is_null($markerwidth) || $markerwidth == 0) {$markerwidth = '50';};?>
                <?php     echo $form->text('markerwidth', $markerwidth);?>
                <span class="input-group-addon">px</span>
            </div>
        </div>
    </div>	
    
    <div class="col-xs-6">
        <div class="form-group">
            <label class="control-label" for="markerheight" name="markerheight">Custom Map Marker Height
                <i class="launch-tooltip fa fa-question-circle" title="<?php     echo t('If you added a custom map marker image above, then you can set a custom Height in pixels for it here. * Enter a number only') ?>"></i>
            </label>
            <div class="input-group">
                <span class="input-group-addon"><i class="fa fa-arrows-v"></i></span>
                <?php     if(is_null($markerheight) || $markerheight == 0) {$markerheight = '50';};?>
                <?php     echo $form->text('markerheight', $markerheight); ?>
                <span class="input-group-addon">px</span>
            </div>
        </div>
    </div>
    
    <div class="col-xs-12">
        <div id="block_note" class="alert alert-info" role="alert"><?php     echo t('Choose from a standard theme option below (Basic)<br><i>or</i><br>Add a custom map style into the textarea for greater control (Advanced).')?></div>
    </div>
    
    <!-- Theme options - Added v0.9.1 -->
    <div class="col-xs-12">
        <div class="form-group">            
            <label class="control-label" for="themehex" name="themehex">Select a standard theme (Basic)
                <i class="launch-tooltip fa fa-question-circle" title="<?php     echo t('To add a standard theme just choose an option from the dropdown list below. If you plan to add custom Snazzy Maps code then make sure to select - Custom (Advanced) - from the list') ?>"></i>
            </label>
            <?php    
                $hexArray = array('ff0000'=>'Red',
                		  'ff7000'=>'Orange',
                	          'ffeb00'=>'Yellow',
                	          '00ff39'=>'Green' ,
                	          '0095ff'=>'Blue' ,
                	          'bd00ff'=>'Purple',
                	          'ff00d4'=>'Pink',
                	          '222222'=>'Greyscale',
                	          'google'=>'Default',
                	          'custom'=>'Custom (Advanced)'
                	           );              
              
            ?>
            <?php     echo $form->select('themehex', $hexArray, $mapObj->themehex);?>
        </div>
    </div><br>
        
	<div class="col-xs-12" id="showHideAdvanced">
            <div class="form-group">               
                <label class="control-label" for="snazzycode" name="snazzycode">Custom Snazzy Maps code (Advanced : Optional)
                    <i class="launch-tooltip fa fa-question-circle" title="<?php     echo t('Use the - Get Snazzy Maps Styles Here - link below to search or create custom styles from the Snazzy Maps website. Once you have the style you would like to use, copy the code and then paste it in here. Make sure the code begins and ends with square brackets [].') ?>"></i>
                </label>                          
                <?php     echo $form->textarea('snazzycode', $mapObj->snazzycode, array('rows' => 9));?>                
            </div> 
            <small><?php     echo t('Get Snazzy Maps styles '); ?><a href="https://snazzymaps.com/" target="_blank"><?php     echo t('Here'); ?></a></small>       
	</div>
  </div><!-- / Row -->  
</div> <!-- / map-styling tab -->

<div id="ccm-tab-content-map-info" class="ccm-tab-content">
  <div class="ccm-google-map-block-container row">  
    <div class="col-xs-12">
        <div id="block_note" class="alert alert-info" role="alert"><?php     echo t('The below option allows you to add a popup info window to your map.<br><br>You have the option to choose if the info window is displayed when the map marker is clicked or when the map loads.<br><br>Great for opening hours, links or any other relevant information.<br><br><i><small>* Leave this blank if you do not require an info window</small></i>')?></div>
    </div>
  
    <div class="col-xs-12">
        <div class="form-group">          
            <label class="control-label" for="info window" name="info window">Info Window Content (Optional)
              <i class="launch-tooltip fa fa-question-circle" 
              title="<?php     echo t('An Info Window displays content (usually text or images) in a popup window above the map.') ?>"></i>
            </label>           
            <?php    
                $editor = Core::make('editor');
                echo $editor->outputStandardEditor('infotext', $mapObj->infotext);
            ?>
         </div>
     </div> 
     
     <div class="col-xs-12">
        <div class="form-group">
            <label class="control-label" for="infostate" name="infostate">
                <?php     echo $form->checkbox('infostate', 1, (is_null($infostate) || $infostate)); ?>
                 Display Info Window When Map Marker Clicked
                 <i class="launch-tooltip fa fa-question-circle" title="<?php     echo t('Check this option if you would like the info window to display only when the map marker is clicked. If unchecked the info window will be displayed when the map is loaded.') ?>"></i>
            </label>
        </div>
    </div>       
  </div><!-- / Row -->
</div> <!-- / map-info tab -->

<script type="text/javascript">
$(function() {
    window.C5GMaps.init();
});
</script>

<script type="text/javascript">
$(document).ready(function(){
    $("select").change(function(){
        $(this).find("option:selected").each(function(){
            if($(this).attr("value")=="custom"){
               
                $("#showHideAdvanced").show();
            }
            else if($(this).attr("value")!="custom"){
                
                $("#showHideAdvanced").hide();
            }
        });
    }).change();
});
</script>